cub/cub/detail/choose_offset.cuh

```diff
@@ -56,6 +56,36 @@ struct ChooseOffsetT
                                          unsigned long long>::type;
 };
 
+template <typename T,typename = void>
+struct is_iterator : std::false_type {};
+template <typename T>
+struct is_iterator<T,std::void_t<typename std::iterator_traits<T>::iterator_category>> : std::true_type {};
+template <typename T>
+inline constexpr bool is_iterator_v = is_iterator<T>::value;
+
+// checks if input type is a valid offset iterator and assigns iterator's value_type to member ::type
+template <typename OffsetIterT>
+struct offset_iter_value {
```

Comment:
This is not SFINAE friendly, I am wondering whether we should reuse `enable_if<Condition, __iter_value_type<OffsetIterT>>`

That siad, I am not sure whether we really need this at all and just go to `__iter_value_type<OffsetIterT>` directly